test/e2e/apps/deployment.go

```diff
@@ -1476,3 +1480,75 @@ func waitForDeploymentCompleteAndCheckRolling(c clientset.Interface, d *appsv1.D
 func waitForDeploymentUpdatedReplicasGTE(c clientset.Interface, ns, deploymentName string, minUpdatedReplicas int32, desiredGeneration int64) error {
 	return testutil.WaitForDeploymentUpdatedReplicasGTE(c, ns, deploymentName, minUpdatedReplicas, desiredGeneration, poll, pollLongTimeout)
 }
+
+//Deployment should have a working scale subresource
+func testDeploymentSubresources(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	deploymentName := "test-new-deployment"
+	podLabels := map[string]string{"name": WebserverImageName}
+	replicas := int32(1)
+	framework.Logf("Creating simple deployment %s", deploymentName)
+	d := e2edeployment.NewDeployment(deploymentName, replicas, podLabels, WebserverImageName, WebserverImage, appsv1.RollingUpdateDeploymentStrategyType)
+	d.Annotations = map[string]string{"test": "should-copy-to-replica-set", v1.LastAppliedConfigAnnotation: "should-not-copy-to-replica-set"}
+	deploy, err := c.AppsV1().Deployments(ns).Create(context.TODO(), d, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Wait for it to be updated to revision 1
+	err = e2edeployment.WaitForDeploymentRevisionAndImage(c, ns, deploymentName, "1", WebserverImage)
+	framework.ExpectNoError(err)
+
+	err = e2edeployment.WaitForDeploymentComplete(c, deploy)
+	framework.ExpectNoError(err)
+
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	newRS, err := deploymentutil.GetNewReplicaSet(deployment, c.AppsV1())
+	framework.ExpectNoError(err)
+	framework.ExpectNotEqual(newRS, nilRs)
```

Comment:
This block is not necessary, it's being used in a different block to ensure the entire deployment state, for this case it's sufficient to only verify completeness from line 1502.

Reply:
Delete complete block